spinalcordtoolbox/vertebrae/detect_c2c3.py

```diff
@@ -0,0 +1,116 @@
+# coding: utf-8
+# This is the interface API to detect the posterior edge of C2-C3 disc.
+# Author: charley
+# Copyright (c) 2018 Polytechnique Montreal <www.neuro.polymtl.ca>
+# About the license: see the file LICENSE.TXT
+
+from __future__ import absolute_import
+
+import os
+import sct_utils as sct
+from sct_flatten_sagittal import flatten_sagittal
+import numpy as np
+import nibabel as nib
+from scipy.ndimage.measurements import center_of_mass
+
+import spinalcordtoolbox.image as msct_image
+from spinalcordtoolbox.image import Image, zeros_like
+
+PATH_MODEL = '/home/charley/Téléchargements/20181128_model_c2c3_t1.yml'
+
+def detect_c2c3(nii_im, nii_seg, contrast, verbose=1):
+    """
+    Detect the posterior edge of C2-C3 disc.
+    :param nii_im:
+    :param nii_seg:
+    :param contrast:
+    :param verbose:
+    :return:
+    """
+    # path to the pmj detector
+    path_sct = os.environ.get("SCT_DIR", os.path.dirname(os.path.dirname(__file__)))
+    path_model = os.path.join(path_sct, 'data', 'c2c3_disc_models', '{}_model'.format(contrast))
+
+    orientation_init = nii_im.orientation
+
+    # Flatten sagittal
+    nii_im = flatten_sagittal(nii_im, nii_seg, centerline_fitting='hanning', verbose=verbose)
+    nii_seg_flat = flatten_sagittal(nii_seg, nii_seg, centerline_fitting='hanning', verbose=verbose)
+
+    # Extract mid-slice
+    nii_im.change_orientation('PIR')
+    nii_seg_flat.change_orientation('PIR')
+    mid_RL = int(np.rint(nii_im.dim[2] * 1.0 / 2))
+    midSlice = nii_im.data[:, :, mid_RL]
+    midSlice_seg = nii_seg_flat.data[:, :, mid_RL]
+    nii_midSlice = msct_image.zeros_like(nii_im)
+    nii_midSlice.data = midSlice
+    nii_midSlice.save('data_midSlice.nii')
```

Comment:
we should do all the i/o in a temporary folder (see `sct.tmp_create()`) to avoid situations where:
- user has a file called data_midSlice.nii in the local folder
- user has writing permission on local folder (e.g. mounted network disk)